lighthouse-core/scripts/internal-analysis/analyze-issues.js

```diff
@@ -0,0 +1,275 @@
+);
+const ISSUES = _ISSUES_SINCE.filter(issue => !issue.pull_request);
+
+/** @param {number} n */
+const percent = n => `${(n * 100).toFixed(1)}%`;
+
+/**
+ * This function only logs *initial* review responses, but could conceivably log all instances of applying "waiting4reviewer".
+ * @param {string} label
+ * @param {Array<AugmentedGitHubIssue>} issues
+ */
+function computeAndLogReviewResponseStats(label, issues) {
+  const initialReviewRequests = issues
+    .map(issue => {
+      const assignEvent = issue.events.find(
+        event => event.event === 'assigned'
+      );
+      const assignee =
+        (assignEvent && assignEvent.assignee && assignEvent.assignee.login) ||
+        '';
+      const firstCommentByAssignee = issue.comments.find(
+        comment => comment.user.login === assignee
+      );
+      const reviewTimeInHours = firstCommentByAssignee
+        ? (new Date(firstCommentByAssignee.created_at).getTime() -
+            new Date(issue.created_at).getTime()) /
+          HOUR_IN_MS
+        : Infinity;
+      return {
+        issue,
+        assignEvent,
+        assignee,
+        firstCommentByAssignee,
+        reviewTimeInHours,
+      };
+    })
+    .filter(review => review.assignee)
+    .sort((a, b) => a.reviewTimeInHours - b.reviewTimeInHours);
+
+  const reviews = initialReviewRequests.filter(
+    review => review.firstCommentByAssignee
+  );
+  const reviewsByLogin = Object.values(
+    initialReviewRequests.reduce(
+      (map, view) => {
+        const reviews = map[view.assignee] || [];
+        reviews.push(view);
+        map[view.assignee] = reviews;
+        return map;
+      },
+      /** @type {Record<string, typeof reviews>} */ ({})
+    )
+  ).sort((a, b) => b.length - a.length);
+
+  const responseTimeInHours = initialReviewRequests.map(r => r.reviewTimeInHours);
+  const medianResponseTime =
+    responseTimeInHours[Math.floor(reviews.length / 2)];
+  console.log(`${log.bold}${label}${log.reset}`);
+  console.log(
+    `  ${percent(
+      initialReviewRequests.length / issues.length
+    )} of PRs requested a review`
+  );
+  console.log(
+    `  ${percent(
+      reviews.length / initialReviewRequests.length
+    )} of requests received a review`
+  );
+  console.log(
+    `  Median initial response time of ${log.bold}${medianResponseTime.toFixed(
+      1
+    )} hours${log.reset}`
+  );
+  console.log('  By User');
+  reviewsByLogin.forEach(requests => {
+    const user = requests[0].assignee;
+    const reviews = requests.filter(r => r.firstCommentByAssignee);
+    const medianResponseTime = requests[Math.floor(reviews.length / 2)].reviewTimeInHours;
+    console.log(
+      `    ${user} - ${requests.length} requests, ${
+        reviews.length
+      } reviews, ${medianResponseTime.toFixed(1)} hours`
+    );
+  });
+}
+
+/**
+ * @param {string} label
+ * @param {Array<AugmentedGitHubIssue>} issues
+ */
+function computeAndLogIssueResponseStats(label, issues) {
```

Comment:
i think its inevitable that we'll want the compute data separately.. so could we split the compute and log bits into separate functions?  (there'll still be a good amount of logic in the log bit, but i think that's fine)

Reply:
I agree, but I'm not sure exactly how to structure the response until we have at least one second usage of it.

WDYT about doing it then?